File: Model/model_TwoState_Alignment.m
function model = model_TwoState_Alignment(ifAlign)
    % an example model to illustrate conservative bound by Schlegel 
    % and Noack. The model is a two-state linear-quadratic system 
    % with negative definte linear part. The system satisfies the TR
    % condition without any coordinate shift. 
    
    % xdot = c + Lx + f(x)
    %   - f(x) is an energy preserving nonlinearity.
    %   - L is choosen as a negative definite symmetric matrix with two
    %     different eigenvalues. L = diag([lam1, lam2]), where lam1 > lam2.
    %   - c is choosen to be ||c|| be a constant with [ifAlign] to the
    %     eigenvector correspond to the largest (least negative)
    %     eigenvalues.

    if nargin < 1
        ifAlign = false;
    end
    
    if ifAlign
        name = "TwoStateBD_Aligned";
        c = [1; 0];
    else
        name = "TwoStateBD_Aligned";
        c = [0; 1];
    end
    
    L = diag([-1, -4]);
    
    Q1 = [0 -1/2; -1/2 0];
    Q2 = [1 0; 0 0];
    Q = cat(3, Q1, Q2);
    
    model = class_Model_LosslessQuad(name, c, L, Q);
end

File: Model/model_TwoState_Alignment.py
import numpy as np
from .class_Model_LosslessQuad import Model_LosslessQuad

def model_TwoState_Alignment(ifAlign: bool = False):
    """Model of a two-state system to illustrate the conservatism of the Schlegel and Noack method.

    Args:
        ifAlign: If True, the constant input c is aligned with the eigenvector corresponding to the larger eigenvalue of L.
    """
    nx = 2
    if ifAlign:
        name = "TwoStateBD_Aligned"
        c = np.zeros((nx, 1))
    else:
        name = "TwoStateBD_NotAligned"
        c = np.array([[0], [1]])

    L = np.diag([-1, -4])

    Q1 = np.array([[0, -0.5], [-0.5, 0]])
    Q2 = np.array([[1, 0], [0, 0]])
    Q = np.stack([Q1, Q2], axis=-1)

    return Model_LosslessQuad(name, c, L, Q)
